fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java

```diff
@@ -287,13 +288,22 @@ private List<Column> getFileSchema() throws DdlException {
             return Lists.newArrayList();
         }
         TNetworkAddress address;
-        List<Long> backendIds = GlobalStateMgr.getCurrentSystemInfo().getBackendIds(true);
-        if (backendIds.isEmpty()) {
-            throw new DdlException("Failed to send proxy request. No alive backends");
+        List<Long> nodeIds = GlobalStateMgr.getCurrentSystemInfo().getBackendIds(true);
+        if (nodeIds.isEmpty()) {
+            if (RunMode.getCurrentRunMode() == RunMode.SHARED_NOTHING) {
+                throw new DdlException("Failed to send proxy request. No alive backends");
+            } else {
+                nodeIds.addAll(GlobalStateMgr.getCurrentSystemInfo().getComputeNodeIds(true));
```

Comment:
in shared-data mode with both BE and CN deployed, this patch will cause the table function always choose BE over CN, which is not expected.